api/base/utils.py

```diff
@@ -22,3 +27,31 @@ def get_object_or_404(model_cls, query_or_pk):
         return model_cls.find_one(query)
     except NoResultsFound:
         raise NotFound
+
+
+def waterbutler_url_for(request_type, provider, path, node_id, token, obj_args=None, **query):
```

Comment:
The one in website.util is more general-purpose than mine, mostly in how it's handling cookies. I'll come back to this when we have CAS integrated, because I think that will affect the differences.